llvm/lib/Target/PowerPC/PPCRegisterInfo.cpp

```diff
@@ -435,7 +435,91 @@ BitVector PPCRegisterInfo::getReservedRegs(const MachineFunction &MF) const {
     }
   }
 
-  assert(checkAllSuperRegsMarked(Reserved));
+  // Mark phony regsiters for the VSR high bits as reserved so that they are
+  // not used.
+  Reserved.set(PPC::FH0);
```

Comment:
Maybe we don't need to reserve these registers since their parent registers are not reserved. If we don't want RA to allocate these registers, better create new register classes for them and these register classes are not allocatable.